Gems/AtomLyIntegration/CommonFeatures/Code/Source/SurfaceData/SurfaceDataMeshComponent.h

```diff
@@ -96,11 +97,12 @@ namespace SurfaceData
 
         // cached data
         AZStd::atomic_bool m_refresh{ false };
-        mutable AZStd::recursive_mutex m_cacheMutex;
+        mutable AZStd::shared_mutex m_cacheMutex;
```

Comment:
I don't recall the specifics, but in the past we've often had to change regular mutexes to recursive_mutexes to prevent it locking up on some platforms (I think iOS and macOS?). No idea if that is the case here, but probably worth checking with Atom and/or platforms that this won't introduce (or re-introduce) some such problem, unless you're sure it's fine.

Reply:
I've emailed the platforms folks to ask. Since this recurses only on read operations, and shared_mutex specifically allows for multi readers, I _think_ it should be fine, but it never hurts to double-check.

Reply:
I don't have a conclusive answer on this one - nobody _knows_ of a problem, but nobody knows if it's fine either. Since I've already previously made this change to many other components, I'm going to submit this change as well. I'll continue to investigate this though. If I discover any problems, I'll need a new PR anyways to back out all the other shared_mutex changes that have already gone in.